File: Learnify.Domain/Entities/Lesson.cs
using System.Text.Json.Serialization;

namespace Learnify.Domain.Entities
{
    public class Lesson : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public Guid ModuleId { get; set; }
        [JsonIgnore] // Add this attribute
        public Module? Module { get; set; }
        public virtual ICollection<File>? Files { get; set; }
    }
}

File: Learnify.UI/src/models/entities/Lesson.js
export class Lesson {
    constructor(
        id,
        title,
        content,
        moduleId,
        module = null,
        files = [],
    ) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.module = module;
        this.moduleId = moduleId;
        this.files = files;
        
    }
}